runtime/vm/jvminit.c

```diff
@@ -1448,6 +1431,79 @@ initializeClassPathEntry (J9JavaVM * javaVM, J9ClassPathEntry *cpEntry)
 	cpEntry->extraInfo = NULL;
 	return CPE_TYPE_UNUSABLE;
 }
+
+IDATA
+initializeModulesPathEntry(J9JavaVM * javaVM, J9ClassPathEntry *cpEntry)
+{
+	PORT_ACCESS_FROM_JAVAVM(javaVM);
+	int32_t attr = 0;
+#if defined(WIN32)
+	char *convertedModulesPath = NULL;
+#endif /* defined(WIN32) */
+
+	/* Start guessing. Is it a directory? */
+	attr = j9file_attr((char *)cpEntry->path);
+	if (EsIsDir == attr) {
+		cpEntry->type = CPE_TYPE_DIRECTORY;
+		return CPE_TYPE_DIRECTORY;
+	}
+
+	if (EsIsFile == attr) {
+		J9JImageIntf *jimageIntf = javaVM->jimageIntf;
+		if (NULL != jimageIntf) {
+			I_32 rc = 0;
+			UDATA jimageHandle = 0;
+			char *modulesPath = (char *)cpEntry->path;
+#if defined(WIN32)
+			IDATA modulesPathLen = cpEntry->pathLength;
```

Comment:
Maybe this needs to be pathLength + 1 to include the terminating NULL, now that the memset has been removed.

Reply:
Discussed with Cheng. He's going to explicitly null terminate the convertedModulesPath, because adding 1 here assume the input is null terminated and we can't assume that.

Reply:
As discussed with Pete offline, I will double-check the updated code by compiling a build locally to see whether it works.
```
				if (convertedModulesPath != NULL) {
					convertedModulesPathLen = j9str_convert(J9STR_CODE_MUTF8, J9STR_CODE_WINDEFAULTACP, modulesPath, modulesPathLen, convertedModulesPath, convertedModulesPathLen);
					if (convertedModulesPathLen > 0) {
					convertedModulesPath[convertedModulesPathLen] = '\0' <------ set up the NULL at the end
					modulesPath = convertedModulesPath;
```

Reply:
Just verified the locally generated build with the code above which works good:
```
C:\JCHAU\测试java>jdk11_openj9_win_fix_v5\bin\java   -version
openjdk version "11.0.9-internal" 2020-10-20
OpenJDK Runtime Environment (build 11.0.9-internal+0-adhoc.Administrator.openj9-openjdk-jdk11master10082020)
Eclipse OpenJ9 VM (build vm_VMInitStages_init_class_path-5b7b7a29d, JRE 11 Windows 10 amd64-64-Bit Compressed References 20201020_000000 (JIT enabled, AOT enabled)
OpenJ9   - 5b7b7a29d
OMR      - b0a570300
JCL      - cf49ddabb9 based on jdk-11.0.9+10)
```